js/element-scrollsnapchanging/demos/carousel/index.html

```diff
@@ -0,0 +1,118 @@
+<!DOCTYPE html>
+<html lang="ru">
+<head>
+  <title>Как работает событие — scrollsnapchanging — Дока</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500&display=swap">
+  <style>
+    html {
+      color-scheme: dark;
+    }
+
+    body {
+      min-height: 100vh;
+      padding: 50px;
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      gap: 50px;
+      background-color: #18191C;
+      color: #FFFFFF;
+      font-family: "Roboto", sans-serif;
+      font-size: 18px;
+    }
+
+    *, *::before, *::after {
+      margin: 0;
+      padding: 0;
+      box-sizing: border-box;
+    }
+
+    .example {
+      width: 100%;
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      align-items: center;
+      gap: 50px;
+    }
+
+    .scroll-container {
+      width: 100%;
+      height: 350px;
+      max-width: 600px;
+      display: flex;
+      overflow-x: scroll;
+      scroll-snap-type: x mandatory;
+    }
+
+    .item {
+      flex: 0 0 70%;
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      height: 100%;
+      background-color: #2E9AFF;
+      color: black;
+      font-size: 2em;
+      scroll-snap-align: center;
+    }
+
+    .item:nth-child(even) {
+      background-color: #123E66;
+      color: #FFFFFF;
+    }
+
+    .caption-value {
+      display: inline-block;
+      width: 20px;
+    }
+
+    @media (width <= 768px) {
+      body {
+        padding: 30px;
+      }
+    }
+  </style>
+</head>
+<body>
+  <figure class="example">
+    <div id="carousel" class="scroll-container">
+      <div class="item">1</div>
+      <div class="item">2</div>
+      <div class="item">3</div>
+      <div class="item">4</div>
+      <div class="item">5</div>
+    </div>
+    <figcaption>
+      <span id="caption-label">Пролистываемый слайд:</span>
+      <span id="caption-value" class="caption-value"></span>
+    </figcaption>
+  </figure>
+  <script>
+    const carousel = document.querySelector('#carousel')
+
+    // Проверяем поддерживает ли браузер событие
+    const supported = 'onscrollsnapchanging' in carousel
+
+    if (supported) {
+      // Добавляем обработчики событий для демки
+      const captionValue = document.querySelector('#caption-value')
+      carousel.addEventListener('scrollsnapchanging', (event) => {
+        captionValue.textContent = event.snapTargetInline.textContent
+      })
+      carousel.addEventListener('scrollend', (event) => {
+        captionValue.textContent = ''
+      })
```

Comment:
Я бы предложил оставить номер слайда. Листаем и смотрим как меняется. Сейчас остановились на слайде 2, а его номер снизу пропал, получается пролистываемый слайд = неизвестно =)

```suggestion
      const captionValue = document.querySelector('#caption-value')

      carousel.addEventListener('scrollsnapchanging', (event) => {
        captionValue.textContent = event.snapTargetInline.textContent
      })
```

Reply:
Сначала так и сделал, но потом подумал, что раз показываем пролистываемый слайд, то надо показывать только в момент прокрутки.

Ок, верну.